firmware/application/ui/ui_geomap.cpp

```diff
@@ -264,8 +264,13 @@ void GeoMapView::focus() {
 void GeoMapView::update_position(float lat, float lon, uint16_t angle) {
 	lat_ = lat;
 	lon_ = lon;
+	
+	// Stupid hack to avoid an event loop
+	geopos.set_report_change(false);
 	geopos.set_lat(lat_);
```

Comment:
Maybe it would be cleaner to add an optional parameter on set_lat and lon, i.e. set_lat(lat_, `bool trigger_change = true`), but I guess is fine as is

Reply:
It looks like in it's current form before this, the update_position method, here and for others apps, would only update the lat value, but the lon would stay the same. Calling set_lat would trigger the report change, which would cause the lon to be set to the current value overriding whatever came in for the update. There was logic (the set_report_change(false)) in other places, but it was missing here.